docs/manual/scala/guide/cluster/code/docs/home/scaladsl/persistence/ShoppingCart.scala

```diff
@@ -0,0 +1,251 @@
+        emptyState = ShoppingCart.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+    // #shopping-cart-create-behavior
+  }
+
+  def createWithTagger(entityContext: EntityContext[Command]): Behavior[Command] = {
+    // #shopping-cart-create-behavior-with-tagger
+    EventSourcedBehavior
+      .withEnforcedReplies[Command, Event, ShoppingCart](
+        persistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId),
+        emptyState = ShoppingCart.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+      .withTagger(AkkaTaggerAdapter.fromLagom(entityContext, Event.Tag))
+    // #shopping-cart-create-behavior-with-tagger
+  }
+
+  def createSnapshots(entityContext: EntityContext[Command]): Behavior[Command] = {
+    // #shopping-cart-create-behavior-with-snapshots
+    EventSourcedBehavior
+      .withEnforcedReplies[Command, Event, ShoppingCart](
+        persistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId),
+        emptyState = ShoppingCart.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+      // snapshot every 100 events and keep at most 2 snapshots on db
+      .withRetention(RetentionCriteria.snapshotEvery(numberOfEvents = 100, keepNSnapshots = 2))
+    // #shopping-cart-create-behavior-with-snapshots
+  }
+
+  // #shopping-cart-apply-behavior-creation
+  def apply(entityContext: EntityContext[Command]): Behavior[Command] = {
+    EventSourcedBehavior
+      .withEnforcedReplies[Command, Event, ShoppingCart](
+        persistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId),
+        emptyState = ShoppingCart.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+      .withTagger(AkkaTaggerAdapter.fromLagom(entityContext, Event.Tag))
+      .withRetention(RetentionCriteria.snapshotEvery(numberOfEvents = 100, keepNSnapshots = 2))
+  }
+  // #shopping-cart-apply-behavior-creation
+
+  implicit val shoppingCartFormat: Format[ShoppingCart] = Json.format
+}
+
+// #shopping-cart-state
+final case class ShoppingCart(
+    items: Map[String, Int],
+    // checkedOutTime defines if cart was checked-out or not:
+    // case None, cart is open
+    // case Some, cart is checked-out
+    checkedOutTime: Option[Instant] = None
+)
+// #shopping-cart-state
+{
+  import ShoppingCart._
+
+  def checkedOut: Boolean = checkedOutTime.nonEmpty
+
+  // #shopping-cart-command-handlers
+  def applyCommand(cmd: Command): ReplyEffect[Event, ShoppingCart] =
+    if (checkedOut) {
+      cmd match {
+        case AddItem(_, _, replyTo) => reply(replyTo)(Rejected("Cannot add an item to a checked-out cart"))
```

Comment:
I would keep the `Effect.reply` to make it clear that this method comes from `Effect`.
```suggestion
        case AddItem(_, _, replyTo) => Effect.reply(replyTo)(Rejected("Cannot add an item to a checked-out cart"))
```